Add attribution to experimental map (#14952)
microsoft/FluidFramework: experimental/dds/attributable-map/README.md

This experimental DDS is a fork of the SharedMap which implements attribution, the plan is to remove it and integrate those APIs into SharedMap.

> ## AttributableMap

Reviewer: I think in practice it's probably easier for people to just leave this called `SharedMap`. That makes macro replace of the import easier for people wanting to try it out.

`SharedMap` is an `EventEmitter`, and will emit events when other clients make modifications. You should register for these events and respond appropriately as the data is modified. `valueChanged` will be emitted in response to a `set` or `delete`, and provide the key and previous value that was stored at that key. `clear` will be emitted in response to a `clear`.

### Attribution

> This experimental DDS allows for the tracking of attribution information, such as the user who made an update and the timestamp of the change. The `attributableMap` currently includes three types of attribution keys: Op-stream, detached, and local.

Reviewer: ```suggestion
This experimental DDS is a copy of `SharedMap` which additionally tracks attribution information, such as the user who made an update and the timestamp of the change.
```
If you want, you could also reference the attributor readme.

> -   The Op-stream attribution key indicates that the update of attribution information is synchronized with the processing of a map operation, and only occurs once the update is ack'd.

Reviewer: I don't think details need to be elaborated on here. We should be documenting expected usage of each attribution key at its declaration rather than in the map readme.
